prov/util/src/util_coll.c

```diff
@@ -1095,6 +1098,8 @@ static int util_av_set_close(struct fid *fid)
 	if (ofi_atomic_get32(&av_set->ref) > 0)
 		return -FI_EBUSY;
 
+	if (av_set->fi_addr_array)
+		free(av_set->fi_addr_array);
```

Comment:
free can accept a null pointer, no need for a separate check

Reply:
Done and submitted as a separate PR as I will need more time to work on the other issues.